### PR TITLE
Cleanup Dockerfile and Makefile

### PR DESCRIPTION
**Pull request by jeethu**
* Changed Makefile to reuse targets
* Removed a few OS level package deps from Dockerfile
* Reordered Dockerfile to exploit layer caching for faster builds

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -1,56 +1,41 @@
...
-ENV         VENV_PIP                $VENV_DIR/bin/pip
-ENV         USER_LOGGING_DIR        /var/log/$USER
-
-
-ADD         requirements.txt /tmp/
+ENV USER     dubizzle
+ENV GROUP    $USER
+ENV USER_DIR /home/$USER
+ENV VENV_DIR $USER_DIR/webapps/flags_env
+ENV REPO_DIR $VENV_DIR/flags VENV_PYTHON $VENV_DIR/bin/python VENV_PIP $VENV_DIR/bin/pip USER_LOGGING_DIR /var/log/$USER
```
Comment (Dareen) [inline docker/Dockerfile]: why merging many declarations in one long line? I think the previous one was easier to read

File: docker/Dockerfile [hunk 2]
```diff
@@ -1,56 +1,41 @@
...
 
 # Install System Packages
 # TODO: filter those packages to only what's needed
```
Comment (Dareen) [inline docker/Dockerfile]: remove this TODO since you've done it :)

File: docker/Dockerfile (hunk 1 above)
Comment (jeethu) [inline docker/Dockerfile]: Each line maps to a layer in the docker image, lesser layers leads to smaller docker images.

File: docker/Dockerfile (hunk 2 above)
Comment (jeethu) [inline docker/Dockerfile]: There are a few more lines to remove.

File: docker/Dockerfile (hunk 1 above)
Comment (Dareen) [inline docker/Dockerfile]: oh i see

**Pull request merged**